apiserver/facades/client/application/application.go

```diff
@@ -1012,30 +1006,6 @@ func (api *APIBase) setCharmWithAgentValidation(
 		return api.applicationSetCharm(ctx, params, newCharm, origin)
 	}
 
-	// Check if the controller agent tools version is greater than the
-	// version we support for the new LXD profiles.
-	// Then check all the units, to see what their agent tools versions is
-	// so that we can ensure that everyone is aligned. If the units version
-	// is too low (i.e. less than the 2.6.0 epoch), then show an error
-	// message that the operator should upgrade to receive the latest
-	// LXD Profile changes.
-
-	// Ensure that we only check agent versions of a charm when we have a
-	// non-empty profile. So this check will only be run in the following
-	// scenarios; adding a profile, upgrading a profile. Removal of a
-	// profile, that had an existing charm, will check if there is currently
-	// an existing charm and if so, run the check.
-	// Checking that is possible, but that would require asking every unit
-	// machines what profiles they currently have and matching with the
-	// incoming update. This could be very costly when you have lots of
-	// machines.
-	if lxdprofile.NotEmpty(lxdCharmProfiler{Charm: currentCharm}) ||
-		lxdprofile.NotEmpty(lxdCharmProfiler{Charm: newCharm}) {
-		if err := validateAgentVersions(ctx, oneApplication, api.modelAgentService); err != nil {
-			return errors.Trace(err)
-		}
-	}
```

Comment:
For readers following along; we didn't support LXD profiles on 2.6 or less, so this check was to ensure that we didn't end up with an agent version that was older.

This can be removed because of this.

Reply:
Thanks